python/ray/_private/runtime_env.py

```diff
@@ -111,18 +112,25 @@ def _xor_bytes(left: bytes, right: bytes) -> bytes:
 
 def _dir_travel(
         path: Path,
-        excludes: Set[Path],
+        excludes: PathSpec,
```

Comment:
This function is starting to get somewhat complicated -- could you add some unit tests for it?